branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/manager/BranchLayoutManager.java

```diff
@@ -0,0 +1,105 @@
+package com.virtuslab.branchlayout.impl.manager;
+
+import java.nio.file.Path;
+
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.collection.List;
+import io.vavr.control.Try;
+import org.checkerframework.checker.index.qual.Positive;
+import org.checkerframework.checker.initialization.qual.UnderInitialization;
+
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutManager;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutReader;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutWriter;
+import com.virtuslab.branchlayout.api.manager.IIndentDefining;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+public class BranchLayoutManager implements IBranchLayoutManager {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("branchLayout");
+
+  private final IBranchLayoutReader reader;
+  private final IBranchLayoutWriter writer;
+
+  public BranchLayoutManager(Path path, boolean isBranchLayoutPresent) {
+    char indentCharacter = BranchLayoutFileUtils.DEFAULT_INDENT_CHARACTER;
+    int indentWidth = BranchLayoutFileUtils.DEFAULT_INDENT_WIDTH;
+    if (isBranchLayoutPresent) {
+      IndentDerivator indentDerivator = new IndentDerivator(path);
+      indentCharacter = indentDerivator.getIndentCharacter();
+      indentWidth = indentDerivator.getIndentWidth();
+    }
+    this.reader = new BranchLayoutFileReader(path, indentCharacter, indentWidth);
+    this.writer = new BranchLayoutFileWriter(path, indentCharacter, indentWidth);
+  }
+
+  @Override
+  public IBranchLayoutReader getReader() {
```

Comment:
`@Getter` for both?

Reply:
I was not possible due to explicit this and delombok issue

Reply:
Btw if `@CheckerSomething` doesn't accept any params,  you should always be able to replace `void foo(@CheckerSomething ThisClass this)` with `@RequiresQualifier(expression = "this", value = CheckerSomething.class) void foo()`

Reply:
(and this shouldn't be removed by Lombok)